### PR TITLE
chore: jq read weight, priority, and preemption from config

### PR DESCRIPTION
**Pull request by hamidzr**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hamidzr** [comment]
related https://github.com/determined-ai/determined/pull/3175

File: master/internal/config/config.go [hunk 1]
```diff
@@ -191,3 +191,106 @@ type InternalConfig struct {
...
+	}
+
+	RMPremption := false // wheter the RM supports preemption
+
+	schdConf := getSchedulerConfig(config, rpName)
+	if preemption := readPreemptionFromScheduler(schdConf); preemption != nil {
+		RMPremption = *preemption
+	}
+
+	// TODO k8 RM?
```
Comment (hamidzr) [inline master/internal/config/config.go:244]: @eecsliu would these need to be read out of help configuration?

Comment (eecsliu) [reply master/internal/config/config.go:267]: You should be able to get these from the `config.ResourceManager.KubernetesRM`. The scheduler is a string, where "preemption" indicates the preemption scheduler. There is no weight, and priority would be 50 by default. K8s doesn't support setting a default priority right now.

File: master/internal/config/config.go [hunk 2]
```diff
@@ -211,3 +211,116 @@ type InternalConfig struct {
...
+
+	// if not found, fall back to resource manager config
+	schedulerConf = config.ResourceManager.AgentRM.Scheduler
+	prio = readPriorityFromScheduler(schedulerConf)
+	if prio != nil {
+		return *prio
+	}
+
+	// TODO k8 RM?
+	return resourcemanagers.DefaultSchedulingPriority
```
Comment (eecsliu) [inline master/internal/config/config.go:310]: You should be able to read the priority from the kubernetes resource manager - the code will be a little different from reading it from the scheduler, but the priority should be up to date and stored in the groups. Default kubernetes priority is 50, though.

File: master/internal/config/config.go (hunk 1 above)
Comment (eecsliu) [reply master/internal/config/config.go:267]: If you want to read whether or not a job is currently being preempted, then we'd need to go to the k8s api. But for this stuff, you can infer from the config options very easily.

**Comment by eecsliu** [comment]
forgot to leave a review comment: looks mostly good - few things k8s related. They should be pretty easy to fix up.

File: master/internal/config/config.go (hunk 2 above)
Comment (hamidzr) [reply master/internal/config/config.go:310]: > but the priority should be up to date and stored in the groups

we should talk about this part and whether we want it to bubble up to the config level to persist. this is currently what happens with adjusting priority I beleive.

**Review by eecsliu — APPROVED** [review]
looks good!

**Comment by eecsliu** [comment]
I'll also talk to you about leftover k8s things

**Comment by hamidzr** [comment]
rebased to resolve the merge conflicts.

**Pull request merged**